### PR TITLE
Fix/broken test behaviours

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

This addresses the remainder of the failures relating primarily to behaviours.

cascade:
base PR #300 <- #302 <- #308

Noteworthy:
- `GenericTransactionBehaviour.teardown` : ba1140929e34c44f561266b1f0908e5d630b47ad
- `BaseSkillTestCase.teardown`: fec958121aede9850c28f447bdd33cf44abce18c

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea hash all --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: packages/fetchai/skills/erc1155_client/tests/test_behaviours.py [hunk 1]
```diff
@@ -52,7 +52,8 @@ def test_setup(self):
     def test_act_is_searching(self):
         """Test the act method of the search behaviour where is_searching is True."""
         # setup
-        self.strategy._is_searching = True
+        self.strategy.is_searching = True
+        assert self.strategy.is_searching
```
Comment (Karrenbelt) [inline packages/fetchai/skills/erc1155_client/tests/test_behaviours.py:56]: the assert I should have removed. I used it to verify prior to setting it directly that this is indeed the issue, there is no private `_is_searching` and property `is_searching`

File: packages/fetchai/skills/generic_buyer/behaviours.py [hunk 2]
```diff
@@ -161,6 +161,11 @@ def _start_processing(self) -> None:
     def teardown(self) -> None:
         """Teardown behaviour."""
 
+        self.processing = None
+        self.processing_time = 0.0
+        self.timedout.clear()
+        self.waiting.clear()
+
```
Comment (Karrenbelt) [inline packages/fetchai/skills/generic_buyer/behaviours.py:168]: please note: I change something other than a test here, this is the only place I did

File: packages/fetchai/skills/generic_buyer/tests/test_behaviours.py [hunk 3]
```diff
@@ -192,70 +203,79 @@ def setup_class(cls):
             ),
         )
 
-    @staticmethod
-    def _check_start_processing_effects(self_, fipa_dialogue, mock_logger) -> None:
+    def setup(self):
```
Comment (Karrenbelt) [inline packages/fetchai/skills/generic_buyer/tests/test_behaviours.py:206]: this was effectively used as a setup, which is what I turned it into

File: packages/fetchai/skills/generic_buyer/tests/test_behaviours.py [hunk 4]
```diff
@@ -285,13 +302,13 @@ def test_act_i(self):
...
         # _start_processing
-        self._check_start_processing_effects(self, fipa_dialogue, mock_logger)
+        self._check_start_processing_effects(self.fipa_dialogue, mock_logger)
 
     def test_act_ii(self):
         """Test the act method of the transaction behaviour where processing is NOT None and processing_time < max_processing."""
         # setup
         processing_time = 5.0
-        self.transaction_behaviour.processing = "some_dialogue"
+        self.transaction_behaviour.processing = "some_dialogue"  # TODO: type
```
Comment (Karrenbelt) [inline packages/fetchai/skills/generic_buyer/tests/test_behaviours.py:311]: there are lots of incorrect types used in test of generic_buyer and generic_seller. This is not urgent or important, but fact of the matter is that there are more `[   ERROR]` in stdout than `PASSED`

File: packages/fetchai/skills/echo/tests/test_handlers.py [hunk 5]
```diff
@@ -61,6 +61,10 @@ def setup_class(cls):
             ),
         )
 
+    def teardown(self) -> None:
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/echo/tests/test_handlers.py:64]: BaseSkillTestCase already has a teardown, why are you overriding it?

File: packages/fetchai/skills/erc1155_client/tests/test_behaviours.py [hunk 6]
```diff
@@ -34,7 +34,7 @@ class TestSearchBehaviour(ERC1155ClientTestCase):
     def test_setup(self):
         """Test the setup method of the search behaviour."""
         # operation
-        self.search_behaviour.setup_class()
+        self.search_behaviour.setup()
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/erc1155_client/tests/test_behaviours.py:37]: Why the change? `search_behaviour` is not a test class...

File: packages/fetchai/skills/erc1155_client/tests/test_behaviours.py (hunk 1 above)
Comment (DavidMinarsch) [inline packages/fetchai/skills/erc1155_client/tests/test_behaviours.py:56]: Looks unrelated, why the change?

File: packages/fetchai/skills/erc1155_deploy/tests/test_behaviours.py [hunk 7]
```diff
@@ -54,7 +54,7 @@ def test_setup(self):
 
         # operation
         with patch.object(self.logger, "log") as mock_logger:
-            self.registration_behaviour.setup_class()
+            self.registration_behaviour.setup()
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/erc1155_deploy/tests/test_behaviours.py:57]: as above

File: packages/fetchai/skills/generic_buyer/tests/test_behaviours.py (hunk 3 above)
Comment (DavidMinarsch) [reply packages/fetchai/skills/generic_buyer/tests/test_behaviours.py:206]: Why are any of the below changes required? I think it could have been achieved with fewer changes

File: packages/fetchai/skills/generic_buyer/tests/test_handlers.py [hunk 8]
```diff
@@ -99,11 +99,6 @@ def setup_class(cls):
             ),
         )
 
-    def teardown(self) -> None:
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_buyer/tests/test_handlers.py:102]: Why removed?

File: packages/fetchai/skills/generic_buyer/tests/test_models.py [hunk 9]
```diff
@@ -41,40 +41,39 @@ class TestGenericStrategy(BaseSkillTestCase):
 
     path_to_skill = PACKAGE_ROOT
 
-    @classmethod
-    def setup_class(cls):
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_buyer/tests/test_models.py:45]: Why are you changing it? What was the problem with it being on the class level?

File: tests/test_test_tools/test_test_skill.py [hunk 10]
```diff
@@ -701,14 +701,6 @@ def test_negative_prepare_skill_dialogue_invalid(self):
                 "counterparty",
             )
 
-    def teardown(self) -> None:
```
Comment (DavidMinarsch) [inline tests/test_test_tools/test_test_skill.py:704]: Why removed?

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: packages/fetchai/skills/erc1155_client/tests/test_behaviours.py (hunk 6 above)
Comment (Karrenbelt) [reply packages/fetchai/skills/erc1155_client/tests/test_behaviours.py:37]: because pycharm search and replace messed this up, it never had a setup_class method to begin with

File: packages/fetchai/skills/erc1155_client/tests/test_behaviours.py (hunk 1 above)
Comment (Karrenbelt) [reply packages/fetchai/skills/erc1155_client/tests/test_behaviours.py:56]: see comment above: https://github.com/valory-xyz/open-aea/pull/308#discussion_r968856718

File: packages/fetchai/skills/generic_buyer/tests/test_models.py (hunk 9 above)
Comment (Karrenbelt) [reply packages/fetchai/skills/generic_buyer/tests/test_models.py:45]: the strategy not being reset. I've changed it now to introduce a teardown where only the strategy gets its original content reset

File: tests/test_test_tools/test_test_skill.py (hunk 10 above)
Comment (Karrenbelt) [reply tests/test_test_tools/test_test_skill.py:704]: resides on base class `BaseSkillTestCase` now

File: packages/fetchai/skills/echo/tests/test_handlers.py (hunk 5 above)
Comment (Karrenbelt) [reply packages/fetchai/skills/echo/tests/test_handlers.py:64]: I introduced that only at the end, removed this now
